registration-processor/registration-processor-notification-service-impl/src/main/java/io/mosip/registration/processor/notification/service/impl/NotificationService.java

```diff
@@ -0,0 +1,530 @@
+import io.mosip.registration.processor.notification.service.dto.MessageSenderDto;
+import io.mosip.registration.processor.notification.service.dto.WorkflowCompletedEventDTO;
+import io.mosip.registration.processor.notification.service.util.StatusNotificationTypeMapUtil;
+import io.mosip.registration.processor.rest.client.audit.builder.AuditLogRequestBuilder;
+import io.mosip.registration.processor.status.code.RegistrationType;
+
+/**
+ * The Class MessageSenderStage.
+ *
+ * @author M1048358 Alok
+ * @since 1.0.0
+ */
+@RefreshScope
+@Service
+public class NotificationService {
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(NotificationService.class);
+
+
+	/** The core audit request builder. */
+	@Autowired
+	private AuditLogRequestBuilder auditLogRequestBuilder;
+
+	/** The notification emails. */
+	@Value("${registration.processor.notification.emails}")
+	private String notificationEmails;
+
+	/** The uin generated subject. */
+	@Value("${registration.processor.uin.generated.subject}")
+	private String uinGeneratedSubject;
+
+	@Value("${registration.processor.uin.activated.subject}")
+	private String uinActivateSubject;
+
+	@Value("${registration.processor.uin.deactivated.subject}")
+	private String uinDeactivateSubject;
+
+	/** The duplicate uin subject. */
+	@Value("${registration.processor.duplicate.uin.subject}")
+	private String duplicateUinSubject;
+
+	/** The reregister subject. */
+	@Value("${registration.processor.reregister.subject}")
+	private String reregisterSubject;
+
+	@Value("${mosip.notificationtype}")
+	private String notificationTypes;
+
+	@Value("${registration.processor.updated.subject}")
+	private String uinUpdatedSubject;
+	
+	@Value("${registration.processor.notification_service_subscriber_hub_url}")
+	private String hubURL;
+	
+	@Value("${registration.processor.notification_service_subscriber_secret}")
+	private String secret;
+	
+	@Value("${registration.processor.notification_service_subscriber_topic}")
+	private String topic;
+	
+	@Value("${registration.processor.notification_service_subscriber_callback_url}")
+	private String callbackURL;
+
+	/** The rest client service. */
+	@Autowired
+	private RegistrationProcessorRestClientService<Object> restClientService;
+
+	/** The service. */
+	@Autowired
+	private MessageNotificationService<SmsResponseDto, ResponseDto, MultipartFile[]> service;
+	
+	@Autowired
+	private SubscriptionClient<SubscriptionChangeRequest,UnsubscriptionRequest, SubscriptionChangeResponse> sb; 
+	
+	@PostConstruct
+	private void init() {
+		SubscriptionChangeRequest subscriptionRequest = new SubscriptionChangeRequest();
+		subscriptionRequest.setCallbackURL(callbackURL);
+		subscriptionRequest.setHubURL(hubURL);
+		subscriptionRequest.setSecret(secret);
+		subscriptionRequest.setTopic(topic);
+		sb.subscribe(subscriptionRequest);
+	}
+
+	@PostMapping(value = "/callback", consumes = "application/json")
```

Comment:
Move the request mappings to a controller layer and have only the business logic in this service